### PR TITLE
Allow to disable the jaeger extension

### LINKED ISSUE #3995: Provides a way to disabled Jaeger opentracing
**Description**
Currently, when you include the `smallrye-opentracing` extension, there is no way to disabled it. But it's very common that, in dev, you don't have a Jaeger server.

Jaeger tracing can also impact the performane of an application so it can be convenient to have an easy way to disabled it for troubleshooting.

The only way to disable the extension is to provides nothing for the `quarkus.jaeger.service-name` config item. But as it generates a WARNING in the log it's not very friendly.

**Implementation ideas**
Add a `quarkus.jaeger.enabled` config item with a default value of `true` so we can disabled it when needed.

As it's  a straightforward change I will provide a PR for it.

### PR DESCRIPTION
**Pull request by loicmathieu**
Fixes #3995

### CODE REVIEW

**Review by objectiser — APPROVED** [review]
Agree about the comment change,

**Review by pavolloffay — APPROVED** [review]
It looks sane. I think this will result in `NoopTracer` being used if jaeger is disabled.

File: extensions/jaeger/runtime/src/main/java/io/quarkus/jaeger/runtime/JaegerConfig.java [hunk 1]
```diff
@@ -16,6 +16,12 @@
 @ConfigRoot(phase = ConfigPhase.RUN_TIME)
 public class JaegerConfig {
 
+    /**
+     * If the Jaeger extension is enabled.
```
[the remark this answers is not in the document]
Comment (loicmathieu) [reply extensions/jaeger/runtime/src/main/java/io/quarkus/jaeger/runtime/JaegerConfig.java]: done

**Review by geoand — APPROVED** [review]
Thanks!

**Pull request merged**